interface/src/components/homepage.js

```diff
@@ -1,8 +1,35 @@
 import React, { Component } from 'react';
-import { Link } from 'react-router-dom'
+import { withRouter } from 'react-router-dom'
+import { getUserData } from './requests'
 import './stylesheets/homepage.css'
 
-class homepage extends Component{
+class Homepage extends Component{
+    constructor(props){
+        super(props);
+        this.updateUser=this.updateUser.bind(this);
+    }
+
+    async updateUser(){
+        let apikey= document.getElementById("api-key").value;
+        if(apikey !== ""){
+            let user= await getUserData(apikey);
+            if(user.hasOwnProperty("error")){
+                return alert('Encountered error: '+ user.error);
+            }
+            if(Object.keys(user).length === 0)
+                return alert('User not found. Please enter a valid API key');
+            let isValid = window.confirm(`Please confirm, are you ${user}?`);
```

Comment:
We do want this confirmation step at that point in time. Basically the user enters their Bety api key, and that can confirm that they put in the correct one because their user name in Bety should show up.

Reply:
OK